Kudu.Contracts/SiteExtensions/ISiteExtensionManager.cs

```diff
@@ -18,6 +18,11 @@ public interface ISiteExtensionManager
         /// </summary>
         Task<SiteExtensionInfo> InstallExtension(string id, string version, string feedUrl);
 
+        /// <summary>
+        /// Before perform a site extension installation, create setting file to keep track of async operation
+        /// </summary>
+        Task<SiteExtensionInfo> InitInstallSiteExtension(string id);
```

Comment:
Does this need to be part of interface contract or simply an implementation detail of InstallExtension?